### PR TITLE
Lock the Accordions for the non superuser

### PR DESCRIPTION
**Pull request by sahsisunny**
# Lock the Accordions for the non superuser

# issue #301 

## What is changes?

- Three files are changed:
  - `user/details/index.html` : Add div and class for every img icon.
  - `user/details/script.js` : Add two functions first one is for lock the accordion and second one is check the user is superuser or not. if user is superuser then it will not lock the accordion. else it will lock the accordion and not load the data of the accordion.
  - `user/details/style.css` : add css for disabled accordion ,and tooltip.
  - `user/images/lock-icon.svg` : add lock icon for disabled accordion.

## After Screenshots
Lock Accordian

### CODE REVIEW

File: user/details/script.js [hunk 1]
```diff
@@ -661,6 +659,46 @@ function showContent() {
...
 }
 
+function ifUserNotSuperUser() {
+  const accordionTabs = document.querySelectorAll('.accordion');
+  const accordionIcon = document.querySelectorAll('.accordion-icon');
+  const toolParent = document.querySelectorAll('.tool-parent');
+  accordionIcon.forEach((icon) => {
+    icon.src = '/user/images/lock-icon.svg';
+    icon.classList.add('accordion-icon');
+    icon.style.width = '60%';
```
Comment (rohan09-raj) [inline user/details/script.js:669]: style property applies an inline style to the selected element, therefore can we please use the existing class to do this?
The drawbacks of inline style here can be if any developers apply the `width` using class or id, they won't be able to do it due to higher specificity of inline styles.

File: user/details/style.css [hunk 2]
```diff
@@ -278,3 +278,32 @@ footer p a {
...
+}
+
+.tool-parent {
+  position: relative;
+}
+
+.tooltip {
+  position: absolute;
+  bottom: 120%;
+  left: -120%;
```
Comment (rohan09-raj) [inline user/details/style.css:294]: Can we please use +ve values here?
Try using `right` property

File: user/details/style.css [hunk 3]
```diff
@@ -278,3 +278,32 @@ footer p a {
...
+}
+
+.tooltip::after {
+  content: '';
+  position: absolute;
+  top: 100%;
+  left: 50%;
+  border-width: 7px;
+  border-style: solid;
+  border-color: black transparent transparent transparent;
```
Comment (rohan09-raj) [inline user/details/style.css:308]: Can we please use shorthands here?

Comment (rohan09-raj) [inline user/details/style.css:308]: Use CSS variable for the colors

File: user/details/script.js [hunk 4]
```diff
@@ -661,6 +659,46 @@ function showContent() {
   section3.classList.remove('hide');
 }
 
+function ifUserNotSuperUser() {
```
Comment (rohan09-raj) [inline user/details/script.js:662]: Function name can be better here, write its name according to the action/usage

**Review by rohan09-raj — CHANGES_REQUESTED** [review]

File: user/details/script.js (hunk 1 above)
Comment (sahsisunny) [reply user/details/script.js:669]: ok i got it

File: user/details/style.css (hunk 2 above)
Comment (sahsisunny) [reply user/details/style.css:294]: same issue to using `right` property
```css
bottom: calc(100% + 10px);
right: calc(50% - 100px);
```
then used above code is it right ?

File: user/details/style.css (hunk 3 above)
Comment (sahsisunny) [reply user/details/style.css:308]: for width and style but color i am not able to use for all property.

File: user/details/script.js (hunk 4 above)
Comment (sahsisunny) [reply user/details/script.js]: done

File: user/details/style.css (hunk 3 above)
Comment (sahsisunny) [reply user/details/style.css]: done

**Review by rohan09-raj — APPROVED** [review]

**Comment by heyrandhir** [comment]
Lock icon should be next to the word itself

File: user/details/script.js [hunk 5]
```diff
@@ -661,6 +659,45 @@ function showContent() {
   section3.classList.remove('hide');
 }
 
+function lockAccordianShowTooltip() {
```
Comment (heyrandhir) [inline user/details/script.js:662]: can we rename the function to `lockAccordiansForNonSuperUser`

File: user/details/script.js [hunk 6]
```diff
@@ -661,6 +659,45 @@ function showContent() {
...
+  const accordionTabs = document.querySelectorAll('.accordion');
+  const accordionIcon = document.querySelectorAll('.accordion-icon');
```
Comment (heyrandhir) [inline user/details/script.js:664]: `accordionIcons`

File: user/details/index.html [hunk 7]
```diff
@@ -23,25 +24,49 @@ <h1>User Management</h1>
           <li class="accordion-academic">
             <div class="visible-content">
               <h2>Professional Details</h2>
-              <img src="../images/arrow-icon.svg" alt="expand" />
+              <div class="tool-parent">
```
Comment (heyrandhir) [inline user/details/index.html:27]: can the class name be something more meaningful

**Review by heyrandhir — CHANGES_REQUESTED** [review]

File: user/details/script.js (hunk 5 above)
Comment (sahsisunny) [reply user/details/script.js]: done, changed

File: user/details/index.html (hunk 7 above)
Comment (sahsisunny) [reply user/details/index.html]: done

File: user/details/script.js (hunk 6 above)
Comment (sahsisunny) [reply user/details/script.js]: changed

File: user/details/style.css [hunk 8]
```diff
@@ -278,3 +278,35 @@ footer p a {
...
+  cursor: none;
+}
+
+.accordion-icon {
+  width: 50%;
+}
+
+.tool-parent {
+  position: relative;
+}
```
Comment (heyrandhir) [inline user/details/style.css:293]: if this is no longer used can we remove it

**Review by heyrandhir — CHANGES_REQUESTED** [review]

File: user/details/style.css (hunk 8 above)
Comment (sahsisunny) [reply user/details/style.css]: changed

**Review by heyrandhir — DISMISSED** [review]
LGTM

**Comment by sahsisunny** [comment]
## After Lock Icon Position Changes
Screenshot 2023-02-02 084623

**Comment by rohan09-raj** [comment]
> Lock icon should be next to the word itself

@heyrandhir My opinion on this would be, the UI looks much better and clean when the lock icons are below each other and right aligned.
image

**Comment by heyrandhir** [comment]
> > Lock icon should be next to the word itself
> 
> @heyrandhir My opinion on this would be, the UI looks much better and clean when the lock icons are below each other and right aligned as there is some symmetry in the UI and eyes can easily follow along the consistent UI image

Hi Rohan . @sahsisunny will confirm with @ankushdharkar. Even I feel the same and agree with your viewpoints.

**Review by heyrandhir — APPROVED** [review]

**Review by rohan09-raj — APPROVED** [review]

**Review by samyakshah3008 — APPROVED** [review]
Great work Sunny, 
Looks good to merge!

**Pull request merged**